### DISCUSSION TITLE
ryanalexmartin build thread.  I'll ask questions in here.

### DISCUSSION THREAD

**Discussion by ryanalexmartin** [corrados/edrumulus#140] [Show and tell]
I recently ordered PCB4 from JLCPCB, with the black solder mask.  It looks great!

Just soldered all the parts and flashed edrumulus to the ESP.

Gotta say, it's looking great:
![IMG_4503](https://github.com/corrados/edrumulus/assets/15975303/627cc035-7965-4f5c-b668-6e39b411276b)
![IMG_4502](https://github.com/corrados/edrumulus/assets/15975303/e9c117d6-e404-46b3-b5c7-405dcc62ffd5)

I'm having some software issues, though:
```
➜  debug git:(main) ./rusemi 38400

RUSEMI - Rust USB Serial to MIDI CLI
Developed by Icaro Ferre

Usage: rusemi [baud_rate]

Using baud rate: 38400

Available ports:
[0] /dev/cu.wlan-debug
[1] /dev/tty.wlan-debug
[2] /dev/cu.BoseQuietComfort35Serie
[3] /dev/tty.BoseQuietComfort35Serie
[4] /dev/cu.Bluetooth-Incoming-Port
[5] /dev/tty.Bluetooth-Incoming-Port
[6] /dev/cu.usbserial-55990074161
[7] /dev/tty.usbserial-55990074161

Enter port number:
6
Selected port: /dev/cu.usbserial-55990074161

Opening port: /dev/cu.usbserial-55990074161
Setting up CoreMIDI virtual ports...
Receiving data on /dev/cu.usbserial-55990074161 at 38400 baud:
Received: [185, 16, 0]
[2024-04-18 14:17:10] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 0]
Received: [153, 48, 1]
[2024-04-18 14:17:10] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:10] NOTE OFF [Ch: 10 | Pitch: 48]
Received: [185, 16, 8]
[2024-04-18 14:17:11] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 8]
Received: [153, 48, 1]
[2024-04-18 14:17:11] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:11] NOTE OFF [Ch: 10 | Pitch: 48]
Received: [185, 16, 0]
[2024-04-18 14:17:12] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 0]
Received: [153, 48, 1]
[2024-04-18 14:17:12] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:12] NOTE OFF [Ch: 10 | Pitch: 48]
Received: [185, 16, 0]
[2024-04-18 14:17:12] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 0]
Received: [153, 48, 1]
[2024-04-18 14:17:12] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:12] NOTE OFF [Ch: 10 | Pitch: 48]
Received: [185, 16, 0]
[2024-04-18 14:17:12] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 0]
Received: [153, 48, 1]
[2024-04-18 14:17:12] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:12] NOTE OFF [Ch: 10 | Pitch: 48]
Received: [185, 16, 0]
[2024-04-18 14:17:12] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 0]
Received: [153, 48, 1]
[2024-04-18 14:17:12] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:12] NOTE OFF [Ch: 10 | Pitch: 48]
Received: [185, 16, 0]
[2024-04-18 14:17:12] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 0]
Received: [153, 48, 1]
[2024-04-18 14:17:12] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:12] NOTE OFF [Ch: 10 | Pitch: 48]
 Received: [185, 16, 7]
[2024-04-18 14:17:13] CONTROL CHANGE [Ch: 10 | CC: 16 | Value: 7]
Received: [153, 48, 1]
[2024-04-18 14:17:13] NOTE ON [Ch: 10 | Pitch: 48 | Vel: 1]
Received: [137, 48, 0]
[2024-04-18 14:17:13] NOTE OFF [Ch: 10 | Pitch: 48]
```

What's the recommended baud rate?  I'm currently using 38400.

I'm getting lots of noise.  Right now I don't have any piezos attached, but I would think that with the complete frontend circuit I shouldn't be getting *any* misfires, let alone this many.  

For right now, I'll try to adjust the sensitivity values in the code.  Increasing the global threshold from 10 to 12 in `edrumulus_parameters.cpp` seems to have fixed this.

Now I'm trying to look into the design of my drums themselves.  Quite naively, I'm directly using a piezo, which seems to be a terrible idea.  Perhaps I should use a capacitor or something next to the piezo inside the actual drum?  Any suggestions?  Maybe my piezo sensors are just poor quality.
![IMG_4504](https://github.com/corrados/edrumulus/assets/15975303/6d22ab24-4279-4279-bd71-ed1cc93d2934)

**Comment by ryanalexmartin** [comment]
It was a grounding issue causing those mis-fires.  Everything seems to be working great now!  No circuit needed on the drums.  A raw piezo works great, I have the values dialed in, and I can start adding other drums now

**Reply by corrados** [reply]
Great to hear that it works now for you :-). Your Edrumulus implementation looks very nice.

Please note that Prototype 4 has a known issue. You need to solder an additional capacity to get rid of some noise spikes, see https://github.com/corrados/edrumulus/blob/main/doc/ChangeLog.md#2022-08-03-edrumulus-prototype-4-raspberry-pi-hat-pcb.

**Reply by ryanalexmartin** [reply]
What's the value of that capacitor?

**Reply by corrados** [reply]
I have used  a 100 nF which works fine for me.

**Reply by ryanalexmartin** [reply]
@corrados It's not clear from the picture which connections need to be jumped.  

Is there a "canonical" prototype board which is the most stable and up-to-date?  If there's a known issue in the prototype board, why is the schematic not updated in hardware?  Please tell me which connection needs to be jumped with the capacitor, and if it's alright, I'll update the KiCad schematic and PCB for prototype 4.

**Reply by corrados** [reply]
> @corrados It's not clear from the picture which connections need to be jumped.

The exact position is not critical. Just "to the 3.3V and GND near the kick drum input".

> Is there a "canonical" prototype board which is the most stable and up-to-date?

No, there isn't. Everything is "alpha" state. If I run Edrumulus together with my PC, I am using Prototype 4. With my Raspberry, I am using Prototype 5.

>  If there's a known issue in the prototype board, why is the schematic not updated in hardware?

My plan is to create a new PCB which uses ESP32-S3. But this will take some time (I have other things to do right now).

**Comment by ryanalexmartin** [comment]
https://www.youtube.com/watch?v=dbzGuadzXY8
I want to use this Arduino nano hi hat implementation with my edrumulus implementation.  Do you think this will play well with edrumulus, or might there be some issues?

I saw that the PCB I printed is intended to be used as a raspberry pi shield, for a hi hat controller.  Where can I find the code/docs for the Raspberry Pi stuff?

**Reply by corrados** [reply]
> Do you think this will play well with edrumulus, or might there be some issues?

If it works with the Roland module, it will most probably also work with Edrumulus.

> I saw that the PCB I printed is intended to be used as a raspberry pi shield, for a hi hat controller. Where can I find the code/docs for the Raspberry Pi stuff?

Initially, I planned to use Prototype 4 together with the Raspberry Pi. But since I created Prototype 5, I have never actually used Prototype 4 together with a Raspberry Pi. I have no idea if that actually works with Prototype 4 since I never tested it. If you really want to use this with a Raspberry Pi, do never connect the USB port of your ESP32 board with a power supply while the hat is sitting on a Raspberry Pi!

**Comment by ryanalexmartin** [comment]
Something happened to my board, and now it's giving me nothing but noise again.  Feeling incredibly frustrated.  I've been trying to get a "stable" implementation for almost 2 years now, but this happens every time I feel like I'm right there on the edge of having a playable drumset.

It's possibly my board was physically damaged.  Nothing was changed in the code to cause this issue to happen.

Could it also be that something in the prototype board is causing overheating or something of the sort?  Am I doing something wrong here?  I've tried to solder/construct this thing multiple times over the past two years.  This always seems to happen.  

Thankfully I ordered multiple PCBs, but the SMDs I ordered are probably not going to be enough to get me through another hardware failure if this happens again.

<img width="651" alt="截圖 2024-04-19 下午6 23 39" src="https://github.com/corrados/edrumulus/assets/15975303/550a61c9-0184-4aad-be00-4c101fbecd06">

**Reply by corrados** [reply]
> Something happened to my board, and now it's giving me nothing but noise again.

What do you mean by that? Do you get lots of false trigger results? That may be caused by noise of the power supply of the USB port on your PC. Maybe try another port or use a powered USB hub (which I do).

**Comment by ryanalexmartin** [comment]
Update:

I was able to get everything working.  I believe the noise was caused by an electrical short erasing some of the data in the microcontroller's flash memory.

After re-flashing the ESP32, things are working again.  My drumset is now (more or less) complete and playable.  I will now make a YouTube video series detailing how to get started with the edrumulus project.

**Reply by corrados** [reply]
That is good news :-). Have you added the capacitor or is it working fine for you without it?

**Comment by ryanalexmartin** [comment]
I’m still dialing in some of the values, especially since all of my drums are 3D printed. After the values are dialed in, I’ll revisit the issue and see if there are random triggers. I’d love to be objective about it, rather than go on “feeling” but I do “feel” the random triggers may be due to physical phenomena rather than electrical (that is to say perhaps sensitivity is set too high, etc).  How did you determine the need for the capacitor?Anyway, here’s the controller and the drumset I have converted. The cymbals and the hi hat controller design need a lot of work, but this is no fault of edrumulus. One thing I would like to see changed is for it to be easier to create custom drums. Right now there are a limited number of commercial products enumerated in edrumulus_parameters.cpp.  I would like to see more generic “templates” for drums so that a larger community of non developers can contribute profiles for all kinds of different drums.  I’ll have to read more about how the profiles work, but I’m thinking I could make a pull request for this… just need to really clearly think it out before I vomit code into the project. 從我的iPhone傳送Volker Fischer ***@***.***> 於 2024年4月24日 04:06 寫道：﻿
That is good news :-). Have you added the capacitor or is it working fine for you without it?

**Reply by corrados** [reply]
> How did you determine the need for the capacitor?

In my setup, I had random MIDI signals with low velocity of the kick input. After soldering the capacitor to the prototype, these false trigger went away.
